src/main/java/school/hei/haapi/endpoint/rest/mapper/FeeMapper.java

```diff
@@ -17,4 +19,25 @@ public Fee toRestFee(school.hei.haapi.model.Fee fee) {
         .creationDatetime(fee.getCreationDatetime())
         .dueDatetime(fee.getDueDatetime());
   }
+
+  public school.hei.haapi.model.Fee toDomainFee(CreateFee createFee) {
+    return school.hei.haapi.model.Fee.builder()
+        .studentId(createFee.getStudentId())
+        .type(toDomainFeeType(createFee.getType()))
+        .totalAmount(createFee.getTotalAmount())
+        .comment(createFee.getComment())
+        .creationDatetime(createFee.getCreationDatetime())
+        .dueDatetime(createFee.getDueDatetime())
+        .build();
+  }
+
+  private Fee.TypeEnum toDomainFeeType(CreateFee.TypeEnum createFeeType) {
+    String feeType = createFeeType.toString();
+    if (feeType.equals(Fee.TypeEnum.TUITION.toString())) {
+      return Fee.TypeEnum.TUITION;
+    } else if (feeType.equals(Fee.TypeEnum.HARDWARE.toString())) {
+      return Fee.TypeEnum.HARDWARE;
+    }
+    throw new BadRequestException("Fee type must be valid");
```

Comment:
`"Unexpected feeType: " + feeType`. We need to see the unexpected value in error logs.